api/datastore/internal/datastoreutil/validator.go

```diff
@@ -31,95 +64,88 @@ func (v *validator) GetApps(ctx context.Context, appFilter *models.AppFilter) ([
 
 // app and app.Name will never be nil/empty.
 func (v *validator) InsertApp(ctx context.Context, app *models.App) (*models.App, error) {
-	if app == nil {
-		return nil, models.ErrDatastoreEmptyApp
-	}
-	if app.Name == "" {
-		return nil, models.ErrDatastoreEmptyAppName
+	err := checkApp(app)
+	if err != nil {
+		return nil, err
 	}
 
 	return v.Datastore.InsertApp(ctx, app)
 }
 
 // app and app.Name will never be nil/empty.
 func (v *validator) UpdateApp(ctx context.Context, app *models.App) (*models.App, error) {
-	if app == nil {
-		return nil, models.ErrDatastoreEmptyApp
-	}
-	if app.Name == "" {
-		return nil, models.ErrDatastoreEmptyAppName
+	err := checkApp(app)
+	if err != nil {
+		return nil, err
 	}
+
 	return v.Datastore.UpdateApp(ctx, app)
 }
 
 // name will never be empty.
-func (v *validator) RemoveApp(ctx context.Context, name string) error {
-	if name == "" {
-		return models.ErrDatastoreEmptyAppName
+func (v *validator) RemoveApp(ctx context.Context, app *models.App) error {
```

Comment:
this should be by id too. (none of these should have 'by name' except for a 'get app [id]' to lookup an app id for an app name.